### ISSUE TITLE
Speedup for Reinhard05

### ISSUE THREAD

**Issue by heckflosse** [LuminanceHDR/LuminanceHDR#92]
I created a [new branch](https://github.com/heckflosse/LuminanceHDR/tree/reinhard05_speedup) in my fork with a speedup for reinhard05

I used a 36 Mpixel file for my tests. Here are the timings:

```
                       before                    after
reinhard05            23200 ms                  550 ms
```

**Comment by Benitoite** [comment]
It is very very fast!
Merged into https://github.com/Benitoite/LuminanceHDR/commits/Test1 and built for mac:https://filebin.net/rycs3cdbfe5gnxrz/Luminance_HDR_2.5.2-SNAPSHOT-Test1g.dmg
![screen shot 2017-12-05 at 9 10 17 am](https://user-images.githubusercontent.com/17919239/33621436-9ce41be2-d99f-11e7-99d5-ae42259a6812.jpg)

**Comment by gaaned92** [comment]
And thanks to @heckflosse  for improvements and @Benitoite  for merging,  W64 users can download

> Luminance-HDR_Test1_v.2.5.1-165-gedf48b5f_WinVista_64.exe

at [my LuminanceHDR drive](https://drive.google.com/open?id=19L6jI85Q_MAXZhQkUBNuEuUrqUqFPxjA)

**Comment by heckflosse** [comment]
@gaaned92 @Hombre57 Thanks to you both for your work on msys2 build which I wasn't able to complete due to lack of knowledge 👍 👍 👍

**Comment by heckflosse** [comment]
Fixed with 0b394f7
